go/libraries/doltcore/branch_control/binlog.go

```diff
@@ -176,36 +221,36 @@ func (row *BinlogRow) Serialize(buffer *bytes.Buffer) {
 	writeUint64(buffer, row.Permissions)
 }
 
-// deserializeBinlogRow returns a BinlogRow from the data at the given position. Also returns the new position. Assumes
-// that the given data's encoded integers are big-endian.
-func deserializeBinlogRow(data []byte, position uint64) (BinlogRow, uint64) {
+// deserializeBinlogRow returns a BinlogRow from the data at the given position. Assumes that the given data's encoded
+// integers are big-endian.
+func deserializeBinlogRow(data []byte, position *uint64) BinlogRow {
```

Comment:
Something to consider: this is a lot of code to maintain in the future when flatbuffers is already widely deployed in this codebase. Did you try a prototype with that?